src/std/text/json/json-test.ss

```diff
@@ -3,37 +3,36 @@
 ;;; :std/text/json unit test
 
 (import :std/test
-        :std/text/json
-        :std/sugar)
+        :std/sugar
+        ./api)
 (export json-test)
 
 (defstruct foo (a b) transparent: #t)
 (defmethod {:write-json foo}
-  (lambda (self port)
+  (lambda (self output)
     (with ((foo a b) self)
-      (write-json-alist [(cons 'b (1+ b)) (cons 'a a)] port))))
+      (write-json (hash (a a) (b b)) output))))
 
-(def json-test
-  (test-suite "test :std/text/json"
-
-    (def (check-encode-decode obj str)
-      (parameterize ((json-sort-keys #f))
-        (check (call-with-input-string (call-with-output-string (cut write-json obj <>)) read-json)
-               => obj))
-      (parameterize ((json-sort-keys #t))
-        (check (call-with-output-string (cut write-json obj <>)) => str)
-        (check (call-with-input-string str read-json) => obj)))
+(def (check-encode-decode obj str)
+  (parameterize ((json-sort-keys #f))
+    (check (call-with-input-string (call-with-output-string (cut write-json obj <>)) read-json)
```

Comment:
Use `json-object->string` and `string->json-object` rather than call-with-foo stuff?

Reply:
It does test the port dispatch this way.

I think we need some more tests here anyway.